src/lib/Components/OpaqueImageView/OpaqueImageView.tsx

```diff
@@ -79,19 +84,22 @@ export default class OpaqueImageView extends React.Component<Props, State> {
   }
 
   imageURL() {
-    const imageURL = this.props.imageURL
+    const { imageURL, disableGemini } = this.props
+
     if (imageURL) {
-      // Either scale or crop, based on if an aspect ratio is available.
-      const type = this.state.aspectRatio ? "fit" : "fill"
-      const width = String(this.state.width)
-      const height = String(this.state.height)
-      // tslint:disable-next-line:max-line-length
-      return `https://${GeminiHost}/?resize_to=${type}&width=${width}&height=${height}&quality=${ImageQuality}&src=${encodeURIComponent(
-        imageURL
-      )}`
-    } else {
-      return null
+      if (disableGemini) {
```

Comment:
Ah I see it here-- curious in which cases this is necessary?

Reply:
The `OpaqueImageView` uses the width+height props you pass into the component to generate the gemini URL. This was a problem for me in situations where the thumbnail in the embedded carousel was a different size than the image in the full screen carousel. It meant that the two images had different URLs and therefore the full screen one would still be loading while the transition into full screen mode happened.

So I decided to optionally decouple gemini url generation to guarantee a cache hit when entering full screen mode.

Reply:
Ah! Maybe the naming is confusing me then-- can we call this bool `resizeImage` or `useDefaultImageUrl` or something like that?

`disableGemini` sounds more like we're talking about an entire system being available/unavailable which threw me off.

Reply:
Renamed to `rawImageURL`, good suggestion! :+1: